chore: bump all major dependencies

SuperFluffy opened the pull request:
## Summary
Bump all major dependencies

## Background

An update to our Astria's dependencies was long overdue. Notable developments have been:

1. penumbra releasing most of their core crates to crates.io (changing the `penumbra-*` prefix to `penumbra-sdk-*`).
2. http and hyper getting stable 1.0 releases
3. celestia-tendermint being replaced by upstream tendermint
4. tendermint having seen a 0.40.0 release with several fixes that we need to establish IBC clients (we were relying on a patched tendermint 0.34 until now).

## Changes

### major dependency upgrades:

1. axum: 0.7.9
2. celestia-types: 0.11.0
3. celestia-rpc: 0.10.0
4. cnidarium: 0.83.0 (instead of upstream rev 686fa5b53)
5. http: 1.2.0
6. hyper: 1.6.0
7. ibc-types: 0.15.0
8. jsonrpsee: 0.24.8
9. metrics: 0.24.1
10. metrics-exporter-prometheus: 0.16.2
11. opentelemetry: 0.27.1
12. opentelemetry-otlp: 0.27.0
13. opentelemetry-semantic-conventions: 0.27.0
14. opentelemetry_sdk: 0.27.1
15. pbjson-build: 0.7.0
16. pbjson-types: 0.7.0
17. penumbra-ibc (renamed to penumbra-sdk-ibc): 1.3.1
18. penumbra-proto (renamed to penumbra-sdk-proto): 1.3.1
19. penumbra-tower-trace (renamed to penumbra-sdk-tower-trace): 1.3.1
20. prost: 0.13.4
21. prost-build: 0.13.4
22. reqwest: 0.12.12
23. tendermint: 0.40.2
24. tendermint-config: 0.40.2
25. tendermint-proto: 0.40.2
26. tendermint-rpc: 0.40.2
27. tonic: 0.12.3
28. tonic-build: 0.12.3
29. tokio: 1.44.2 (to address https://rustsec.org/advisories/RUSTSEC-2025-0023 while we are it)
30. tower: 0.5.2 (NOTE: penumbra is still using tower 0.4, but at this point sequencer's use of tower and its penumbra dependency don't interact so there are clashes)
31. tower-http: 0.6.2
32. tracing-opentelemetry: 0.28.0

### removed dependencies

1. celestia-tendermint: celestia-types
now relies on upstream tendermint and
celestia-tendermint is no longer maintained.
2. opentelemetry-stdout: not to be used in
production code and the APIs we used were removed.
3. isahc: replaced by reqwest because it has not
seen updates in a long time and was not updated
to the work with hyper and http 1.0 stabilization.
4. http-body and http-body-util (not needed; can use tonic reexports).

### Other and crate/service specific changes

Several crates that had a custom defined
dependency now use the workspace definition.

All services lost their `<SERVICVE_NAME>_PRETTY_PRINT`
env var, because it was no longer necessary. See the
telemetry changes below.

#### Conductor

celestia-types no longer has celestia-tendermint
as a dependency and instead uses tendermint 0.40
(the same as Astria uses). This required changing
a bunch imports and type paths.

`Blob::new` now requires an `AppVersion` argument
(only needed in tests. Mainnet is running v3.2.0,
so the blackbox tests sets the same, but the
conductor business logic is oblivious to that).

Type `celestia_types::ataAvailabilityHeader` no longer
has public fields and instead provides a constructor
(only in tests, Conductor business logic is again oblivious).

From celestia-node v0.15.0 blob.GetAll no longer
returns an error if a single requested namespace
(out of a range of namespaces) does not contain
blobs. Instead, the namespace does not show up in
the list of blobs. If there are no blobs at all
then the RPC omits the field entirely, which
the Rust celestia-rpc crate represents as a
`Option<Vec<Blob>>`. All code related to
mapping errors like `BlobNotFound` to an empty
vec have been removed. The current version of
celestia-node on Astria's mainnet is 0.20.0.

tower introduced the BoxCloneSyncService, which
makes constructing a rate-limited, buffered
service a bit easier.

notable sequencer-client changes:

changes except for tests worked out of the box,
but tokio for some reason was a direct dependency.
Changed it to derive from the workspace.

tendermint_rpc::endpoint::broadcast::Response
has a new field codespace, which needed to be
populated in test code.

#### Composer

the api server needed refactoring because axum
had breaking changes. To avoid dealing with
the type parameters in the new axum::serve::Serve
future, composer turns it into a trait object and
wraps it in a api::Serve future.

The geth related blackbox tests also started
to be more flaky for unknown reasons - extending
the timeout duration for the tests mostly fixes
the flakiness. But this will have to be
be investigated in a follow-up PR.

#### Bridge-Withdrawer

axum::Server was replaced by axum::serve::Serve,
which requires a tokio::net::TcpListener to be
instantiated. We have introduced `api::Serve` as a
type-erased wrapper to `axum::serve::Serve` to
avoid having to specify its type parameteres while
still providing access to its bound socket.
However, because of these changes
BridgeWithdrawer::new is now async.

the http dependency is now specified through the
workspace.

The shutdown signal for the API server was also
changed to be a CancellationToken.

#### Sequencer-Relayer

axum underwent heavy changes and has replaced
axum::Server by axum::serve::Serve future. This
patch wraps the future to avoid having to name its
type, erasing it, while still giving access to the
bound tcp socket address. Because its constructor
takes a tokio::net::TcpListener, the API server
setup is now async, which in turn required
SequencerRelayer::new to become async.
Due to type erasure + wrapping, the shutdown
signal closure now has to be passed in early on
instead of right before entering
sequencer-relayer's runloop. While doing so,
the oneshot channel was replaced by a cancellation
token.

Because Celestia blobs now require an app version,
the relayer write path now sets it to version 3
(because astria mainnet is targetting celestia-app
3.2.0, and the consensus node is targetting
3.3.0).

In tests, isahc was replaced by reqwest. The
entire workspace is migrating to hyper 1+ and
http 1+ and isahc has not seen upgrades in several
months and there is no work underway to make it
compatible with http 1+.

dependency tower was removed because it was not
used.

#### Auctioneer

there was no need for the direct dependencies
on http-body and http-body-util because tonic
already provided reexports and/or type aliases.

#### Telemetry

opentelemetry had a couple of breaking changes
that required reworking how the otel pipeline
is set up.

the most visible change is opentelemetry-stdout
no longer allowing to write to stderr or another
sink. because the otel stdout writer was never
used and because opentelemetry-stdout itself
notes that is should not be used in production
contexts this patch removes it entirely (and
with it the the need to explicitly set pretty
printed logs; if services are connected to a
TTY or the FORCE_STDOUT flag is set, they will
always write to stdout).



## Testing
Only dependencies were upgraded. If tests and smoke tests still work then nothing should be broken.

## Changelogs
No updatates required.


## Related Issues
closes #2094, #2064

Fraser999 reviewed (approved):
LGTM - nice job.  Approval subject to @noot confirming the penumbra changes in the top-level Cargo.toml are valid.

SuperFluffy commented on crates/astria-conductor/Cargo.toml:
```diff
@@ -46,26 +46,34 @@ thiserror = { workspace = true }
 tokio = { workspace = true, features = ["macros", "rt-multi-thread", "signal"] }
 tokio-util = { workspace = true, features = ["rt"] }
 tonic = { workspace = true, features = ["tls", "tls-roots"] }
+# tower = { workspace = true, features = ["buffer", "limit"] }
```
Yes, thank you for noticing. Done in 81ab94c9

SuperFluffy commented on crates/astria-conductor/Cargo.toml:
```diff
@@ -46,26 +46,34 @@ thiserror = { workspace = true }
+# Astria and Celestia both. If there is a version mismatch, the version
+# required for celestia needs to be imported under an alias.
+celestia-rpc = "0.9.0"
+# celestia-tendermint = { package = "tendermint", "0.40.0" }
```
No - it's explained in the long comment above. But because that's not clear I have added a single comment above this line to explain the reasoning (it's for when celestia-rpc and celestia-types lag behind upstream tendermint).

81ab94c9

SuperFluffy commented on crates/astria-sequencer-relayer/src/relayer/write/conversion.rs:
```diff
@@ -137,7 +137,10 @@ impl Payload {
         let encoded = value.encode_to_vec();
         let compressed = compress_bytes(&encoded)?;
-        let blob = Blob::new(namespace, compressed)?;
+        // TODO(janis): should sequencer-relayer query its celestia endpoint to dermine which
```
I have made https://github.com/astriaorg/astria/issues/2092 and will add a permalink to this line to the issue once we have this merged.

SuperFluffy commented:
I bumped tokio to 1.44.2 in response to https://rustsec.org/advisories/RUSTSEC-2025-0023, updated celestia-types to 0.11, celestia-rpc to 0.10, and undid a change to astria-telemetry: tests were extra noisy because I removed the ability to send events to `std::io::sink` (went overboard with removing the otlp stdout logic).

joroshiba commented:
> TODO: @joroshiba you pointed out some relevant patches we need?

Noting on this from the PR note that 0.40 has all the updates we require, patch was only needed for older version and it was pulling in an upstream fix.

ethanoroshiba commented on crates/astria-conductor/src/celestia/verify.rs:
```diff
@@ -498,43 +495,38 @@ impl RateLimitedVerificationClient {
-                        } => fetch_validators_with_retry(client, prev_height, height).await,
-                    }
+        let service_builder = tower::ServiceBuilder::new()
+            // XXX: This number is arbitarily set to the same number os the rate-limit. Does that
```
```suggestion
            // XXX: This number is arbitarily set to the same number as the rate-limit. Does that
```

ethanoroshiba commented on crates/astria-conductor/src/celestia/verify.rs:
```diff
@@ -498,43 +495,38 @@ impl RateLimitedVerificationClient {
-                    }
+        let service_builder = tower::ServiceBuilder::new()
+            // XXX: This number is arbitarily set to the same number os the rate-limit. Does that
+            // make sense? Should the number be set higher?
```
Maybe? It does seem a bit odd to have the buffer be the same size as requests per second, since we'd essentially allow the buffer to be completely filled in one second if it were to reach the rate limit. Practically, I'm not sure if this ever even happens though, so it would certainly be helpful if we had some metrics around this or something. If we set the buffer higher without having these numbers it seems a bit arbitrary either way IMO.

ethanoroshiba reviewed (approved):
LGTM on Conductor end :)

bharath-123 reviewed (approved):
I have reviewed the changes in composer and auctioneer. And they LGTM!
